packages/core/src/endowmentsToolkit.js

```diff
@@ -172,13 +168,65 @@ function endowmentsToolkit({
   }
 
   /**
-   * @param {string} visitedPath
-   * @param {string[]} pathParts
-   * @param {string[]} explicitlyBanned
-   * @param {object} sourceRef
-   * @param {object} targetRef
-   * @param {object} unwrapTo
-   * @param {object} unwrapFrom
+   * @template {object} T
+   * @param {T} moduleNamespace
+   * @param {string} moduleId
+   * @param {LMPolicy.BuiltinPolicy} policyBuiltin
+   * @returns {Partial<T>}
+   */
+  function getBuiltinForConfig(moduleNamespace, moduleId, policyBuiltin) {
+    /** @type {string[]} */
+    const builtinPaths = []
+
+    /** @type {string[]} */
+    const explicitlyBanned = []
+
+    // Collect the same paths information as getEndowmentsForConfig to enable
+    // matching behavior of policy between globals and builtins
+    Object.entries(policyBuiltin).forEach(([packagePath, allowed]) => {
+      const packagePathParts = packagePath.split('.')
+      if (moduleId === packagePathParts[0]) {
+        const packagePathWithoutPackage = packagePathParts.slice(1).join('.')
+        if (allowed === true) {
+          builtinPaths.push(packagePathWithoutPackage)
+        } else if (allowed === false) {
+          explicitlyBanned.push(packagePathWithoutPackage)
+        }
+      }
+    })
+    const moduleNamespaceView = makeMinimalViewOfRef(
+      moduleNamespace,
+      builtinPaths.sort(),
+      undefined,
+      undefined,
+      explicitlyBanned
+    )
+    return moduleNamespaceView
+  }
+
+  /**
+   * @param {string} visited
+   * @param {string} next
+   */
+  function extendPath(visited, next) {
+    // FIXME: second part of this conditional should be unnecessary
+    if (!visited || visited.length === 0) {
```

Comment:
Just remove it, imo.

```suggestion    
    if (!visited) {
```

Reply:
It was there before and we didn't know why. FIXME was added in a conversation where we weren't confident enough to remove it. I don't want to do it as part of this PR.